pagerduty/resource_pagerduty_extension_servicenow_test.go

```diff
@@ -0,0 +1,193 @@
+package pagerduty
+
+import (
+	"fmt"
+	"log"
+	"strings"
+	"testing"
+
+	"github.com/hashicorp/terraform-plugin-sdk/helper/resource"
+	"github.com/hashicorp/terraform-plugin-sdk/terraform"
+	"github.com/heimweh/go-pagerduty/pagerduty"
+)
+
+func init() {
+	resource.AddTestSweepers("pagerduty_extension_servicenow", &resource.Sweeper{
+		Name: "pagerduty_extension_servicenow",
+		F:    testSweepExtension,
+	})
+}
+
+func testSweepExtensionServicenow(region string) error {
+	config, err := sharedConfigForRegion(region)
+	if err != nil {
+		return err
+	}
+
+	client, err := config.Client()
+	if err != nil {
+		return err
+	}
+
+	resp, _, err := client.Extensions.List(&pagerduty.ListExtensionsOptions{})
+	if err != nil {
+		return err
+	}
+
+	for _, extension := range resp.Extensions {
+		if strings.HasPrefix(extension.Name, "test") || strings.HasPrefix(extension.Name, "tf-") {
+			log.Printf("Destroying extension %s (%s)", extension.Name, extension.ID)
+			if _, err := client.Extensions.Delete(extension.ID); err != nil {
+				return err
+			}
+		}
+	}
+
+	return nil
+}
+
+func TestAccPagerDutyExtensionServicenow_Basic(t *testing.T) {
+	extension_name := resource.PrefixedUniqueId("tf-")
+	extension_name_updated := resource.PrefixedUniqueId("tf-")
+	name := resource.PrefixedUniqueId("tf-")
+	url := "https://example.com/recieve_a_pagerduty_webhook"
+	url_updated := "https://example.com/webhook_foo"
+
+	resource.Test(t, resource.TestCase{
+		PreCheck:     func() { testAccPreCheck(t) },
+		Providers:    testAccProviders,
+		CheckDestroy: testAccCheckPagerDutyExtensionDestroy,
+		Steps: []resource.TestStep{
+			{
+				Config: testAccCheckPagerDutyExtensionServicenowConfig(name, extension_name, url, "false", "any"),
+				Check: resource.ComposeTestCheckFunc(
+					testAccCheckPagerDutyExtensionServicenowExists("pagerduty_extension_servicenow.foo"),
+					resource.TestCheckResourceAttr(
+						"pagerduty_extension_servicenow.foo", "name", extension_name),
+					resource.TestCheckResourceAttr(
+						"pagerduty_extension_servicenow.foo", "extension_schema", "PJFWPEP"),
+					resource.TestCheckResourceAttr(
+						"pagerduty_extension_servicenow.foo", "endpoint_url", url),
+					resource.TestCheckResourceAttr(
+						"pagerduty_extension_servicenow.foo", "config", "{\"notify_types\":{\"acknowledge\":false,\"assignments\":false,\"resolve\":false},\"restrict\":\"any\"}"),
```

Comment:
There is no longer a `config` attribute in this resource. Please break out these checks to match the current schema of the resource.